dask/dataframe/groupby.py

```diff
@@ -610,12 +627,87 @@ def _aca_agg(self, token, func, aggfunc=None, split_every=None, split_out=1):
 
         return aca([self.obj, self.index] if not isinstance(self.index, list) else [self.obj] + self.index,
                    chunk=_apply_chunk,
-                   chunk_kwargs=dict(func=func, columns=columns),
+                   chunk_kwargs=dict(chunk=func, columns=columns),
                    aggregate=_groupby_aggregate,
                    meta=meta, token=token, split_every=split_every,
                    aggregate_kwargs=dict(aggfunc=aggfunc, levels=levels),
                    split_out=split_out, split_out_setup=split_out_on_index)
 
+    def _cum_agg(self, token, chunk, aggregate, neutral):
+        """ Wrapper for cumulative groupby operation """
+        meta = chunk(self._meta)
+        columns = meta.name if isinstance(meta, pd.Series) else meta.columns
+        index = self.index if isinstance(self.index, list) else [self.index]
+
+        # cumulate each partitions
+        name1 = '{0}{1}-map'.format(self._token_prefix, token)
+        cumpart_raw = map_partitions(_apply_chunk, self.obj, *index,
+                                     chunk=chunk,
+                                     columns=columns,
+                                     token=name1,
+                                     meta=meta)
+        cumpart_ext = (cumpart_raw.to_frame()
+                       if isinstance(meta, pd.Series)
+                       else cumpart_raw).assign(**{i: self.obj[i]
+                                                   for i in index})
+
+        name2 = '{0}{1}-take-last'.format(self._token_prefix, token)
+        cumlast = map_partitions(_apply_chunk, cumpart_ext, *index,
+                                 columns=0 if columns is None else columns,
+                                 chunk=M.last,
+                                 meta=meta,
+                                 token=name2)
+
+        name = '{0}{1}'.format(self._token_prefix, token)
+        cname = '{0}{1}-cum-last'.format(self._token_prefix, token)
```

Comment:
Perhaps instead of all the calls to format:

```python
name = self._token_prefix + token
name1 = name + '-map'
name2 = name + '-take-last'
cname = name + '-cum-last'
```

I think this is slightly more readable. Also, slight request for more meaningful variable names.